snapcraft/internal/lxd/_errors.py

```diff
@@ -0,0 +1,176 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2018 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+from typing import List  # noqa
+
+from snapcraft.internal.errors import SnapcraftError
+
+
+logger = logging.getLogger(__name__)
+
+
+class ContainerError(SnapcraftError):
+    """Base class for all container-related exceptions.
+
+    :cvar fmt: A format string that daughter classes override
+
+    """
+
+
+class ContainerArchitectureError(ContainerError):
+
+    fmt = (
+        "Failed to detect container architecture: "
+        "The output from 'lxc info' could not be read:\n{lxc_info}"
+    )
+
+    def __init__(self, lxc_info: str) -> None:
+        super().__init__(lxc_info=lxc_info)
+
+
+class ContainerLXDNotInstalledError(ContainerError):
+
+    fmt = (
+        "Failed to initialize container: "
+        "LXD could not be found. "
```

Comment:
There should be a newline here, not on the following line. (what: why\nhow to fix)